File: packages/csharp-sdk/src/Wusul/Models/AccessPass.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Wusul.Models
{
    /// <summary>
    /// Access Pass model
    /// </summary>
    public class AccessPass
    {
        [JsonProperty("id")]
        public string Id { get; set; } = string.Empty;

        [JsonProperty("cardTemplateId")]
        public string CardTemplateId { get; set; } = string.Empty;

        [JsonProperty("employeeId")]
        public string? EmployeeId { get; set; }

        [JsonProperty("tagId")]
        public string? TagId { get; set; }

        [JsonProperty("siteCode")]
        public string? SiteCode { get; set; }

        [JsonProperty("cardNumber")]
        public string? CardNumber { get; set; }

        [JsonProperty("fileData")]
        public string? FileData { get; set; }

        [JsonProperty("fullName")]
        public string FullName { get; set; } = string.Empty;

        [JsonProperty("email")]
        public string? Email { get; set; }

        [JsonProperty("phoneNumber")]
        public string? PhoneNumber { get; set; }

        [JsonProperty("classification")]
        public string? Classification { get; set; }

        [JsonProperty("startDate")]
        public string StartDate { get; set; } = string.Empty;

        [JsonProperty("expirationDate")]
        public string ExpirationDate { get; set; } = string.Empty;

        [JsonProperty("employeePhoto")]
        public string? EmployeePhoto { get; set; }

        [JsonProperty("title")]
        public string? Title { get; set; }

        [JsonProperty("state")]
        public string State { get; set; } = "active";

        [JsonProperty("url")]
        public string? Url { get; set; }

        [JsonProperty("metadata")]
        public Dictionary<string, object>? Metadata { get; set; }

        [JsonProperty("createdAt")]
        public string CreatedAt { get; set; } = string.Empty;

        [JsonProperty("updatedAt")]
        public string UpdatedAt { get; set; } = string.Empty;
    }

    /// <summary>
    /// Request model for issuing an access pass
    /// </summary>
    public class IssueAccessPassRequest
    {
        [JsonProperty("cardTemplateId")]
        public string CardTemplateId { get; set; } = string.Empty;

        [JsonProperty("employeeId")]
        public string? EmployeeId { get; set; }

        [JsonProperty("tagId")]
        public string? TagId { get; set; }

        [JsonProperty("siteCode")]
        public string? SiteCode { get; set; }

        [JsonProperty("cardNumber")]
        public string CardNumber { get; set; } = string.Empty;

        [JsonProperty("fileData")]
        public string? FileData { get; set; }

        [JsonProperty("fullName")]
        public string FullName { get; set; } = string.Empty;

        [JsonProperty("email")]
        public string? Email { get; set; }

        [JsonProperty("phoneNumber")]
        public string? PhoneNumber { get; set; }

        [JsonProperty("classification")]
        public string? Classification { get; set; }

        [JsonProperty("startDate")]
        public string StartDate { get; set; } = string.Empty;

        [JsonProperty("expirationDate")]
        public string ExpirationDate { get; set; } = string.Empty;

        [JsonProperty("employeePhoto")]
        public string? EmployeePhoto { get; set; }

        [JsonProperty("title")]
        public string? Title { get; set; }

        [JsonProperty("metadata")]
        public Dictionary<string, object>? Metadata { get; set; }
    }

    /// <summary>
    /// Request model for updating an access pass
    /// </summary>
    public class UpdateAccessPassRequest
    {
        [JsonProperty("employeeId")]
        public string? EmployeeId { get; set; }

        [JsonProperty("fullName")]
        public string? FullName { get; set; }

        [JsonProperty("classification")]
        public string? Classification { get; set; }

        [JsonProperty("expirationDate")]
        public string? ExpirationDate { get; set; }

        [JsonProperty("employeePhoto")]
        public string? EmployeePhoto { get; set; }

        [JsonProperty("title")]
        public string? Title { get; set; }

        [JsonProperty("fileData")]
        public string? FileData { get; set; }

        [JsonProperty("metadata")]
        public Dictionary<string, object>? Metadata { get; set; }
    }

    /// <summary>
    /// Request model for listing access passes
    /// </summary>
    public class ListAccessPassesRequest
    {
        [JsonProperty("templateId")]
        public string? TemplateId { get; set; }

        [JsonProperty("state")]
        public string? State { get; set; }
    }
}

File: packages/java-sdk/src/main/java/io/wusul/sdk/models/AccessPass.java
package io.wusul.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

/**
 * Access Pass model.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AccessPass {
    private String id;

    @JsonProperty("cardTemplateId")
    private String cardTemplateId;

    @JsonProperty("employeeId")
    private String employeeId;

    @JsonProperty("tagId")
    private String tagId;

    @JsonProperty("siteCode")
    private String siteCode;

    @JsonProperty("cardNumber")
    private String cardNumber;

    @JsonProperty("fileData")
    private String fileData;

    @JsonProperty("fullName")
    private String fullName;

    private String email;

    @JsonProperty("phoneNumber")
    private String phoneNumber;

    private Classification classification;

    @JsonProperty("startDate")
    private String startDate;

    @JsonProperty("expirationDate")
    private String expirationDate;

    @JsonProperty("employeePhoto")
    private String employeePhoto;

    private String title;
    private AccessPassState state;
    private String url;
    private Map<String, Object> metadata;

    @JsonProperty("createdAt")
    private String createdAt;

    @JsonProperty("updatedAt")
    private String updatedAt;

    // Getters and setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getCardTemplateId() { return cardTemplateId; }
    public void setCardTemplateId(String cardTemplateId) { this.cardTemplateId = cardTemplateId; }

    public String getEmployeeId() { return employeeId; }
    public void setEmployeeId(String employeeId) { this.employeeId = employeeId; }

    public String getTagId() { return tagId; }
    public void setTagId(String tagId) { this.tagId = tagId; }

    public String getSiteCode() { return siteCode; }
    public void setSiteCode(String siteCode) { this.siteCode = siteCode; }

    public String getCardNumber() { return cardNumber; }
    public void setCardNumber(String cardNumber) { this.cardNumber = cardNumber; }

    public String getFileData() { return fileData; }
    public void setFileData(String fileData) { this.fileData = fileData; }

    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPhoneNumber() { return phoneNumber; }
    public void setPhoneNumber(String phoneNumber) { this.phoneNumber = phoneNumber; }

    public Classification getClassification() { return classification; }
    public void setClassification(Classification classification) { this.classification = classification; }

    public String getStartDate() { return startDate; }
    public void setStartDate(String startDate) { this.startDate = startDate; }

    public String getExpirationDate() { return expirationDate; }
    public void setExpirationDate(String expirationDate) { this.expirationDate = expirationDate; }

    public String getEmployeePhoto() { return employeePhoto; }
    public void setEmployeePhoto(String employeePhoto) { this.employeePhoto = employeePhoto; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public AccessPassState getState() { return state; }
    public void setState(AccessPassState state) { this.state = state; }

    public String getUrl() { return url; }
    public void setUrl(String url) { this.url = url; }

    public Map<String, Object> getMetadata() { return metadata; }
    public void setMetadata(Map<String, Object> metadata) { this.metadata = metadata; }

    public String getCreatedAt() { return createdAt; }
    public void setCreatedAt(String createdAt) { this.createdAt = createdAt; }

    public String getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
}
